Statistics/Distributions/ContinuousDistribution.cs

```diff
@@ -36,79 +36,81 @@ public IDistribution Sample(double[] packetOfRandomNumbers)
-                            else if (ele.Attribute(sa.Name).Value == "-Infinity")
-                            {
-                                valD = double.NegativeInfinity;
-                            }
-                            else
-                            {
-                                valD = Convert.ToDouble(ele.Attribute(sa.Name).Value);
-                            }
-                            pi.SetValue(dist, valD);
-                            break;
-                        case "Boolean":
-                            bool valB = Convert.ToBoolean(ele.Attribute(sa.Name).Value);
-                            pi.SetValue(dist, valB);
-                            break;
-                        case "Int32":
-                            int vali = Convert.ToInt32(ele.Attribute(sa.Name).Value);
-                            pi.SetValue(dist, vali);
-                            break;
-                        default:
-                            throw new ArgumentException("unrecognized type in serialization " + sa.type.Name);
-                    }
+                        switch (storedAttribute.type.Name)
+                        {
+                            case "double":
+                                double vald = 0.0; //TODO: WHAT DOES VALD MEAN? 
+                                if (xElement.Attribute(storedAttribute.Name).Value == "Infinity")
+                                {
+                                    vald = double.PositiveInfinity;
+                                }
+                                else if (xElement.Attribute(storedAttribute.Name).Value == "-Infinity")
+                                {
+                                    vald = double.NegativeInfinity;
+                                }
+                                else
+                                {
+                                    vald = Convert.ToDouble(xElement.Attribute(storedAttribute.Name).Value);
+                                }
+                                propertyInfo.SetValue(iDistribution, vald);
+                                break;
+                            case "Double":
+                                double valD = 0.0;
+                                if (xElement.Attribute(storedAttribute.Name).Value == "Infinity")
+                                {
+                                    valD = double.PositiveInfinity;
+                                }
+                                else if (xElement.Attribute(storedAttribute.Name).Value == "-Infinity")
+                                {
+                                    valD = double.NegativeInfinity;
+                                }
+                                else
+                                {
+                                    valD = Convert.ToDouble(xElement.Attribute(storedAttribute.Name).Value);
+                                }//TODO: WHAT ARE THE VAL D VAL B VAL I IT IS NOT CLEAR 
```

Comment:
value Boolean value Double. 
We are taking the value off of an xElement - for a given attribute for a given name... we are switching on a type, so we are combining a string representation of a value with a type, value <type> instead of writing Double and Boolean and double, and Integer and int32 etc. I abbreviated the type, because the case of the type is presented in the switch case to provide immediate context.